paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/SafeOreBreak.java

```diff
@@ -0,0 +1,199 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks.basic;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHack;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHackConfig;
+import com.programmerdan.minecraft.simpleadminhacks.framework.autoload.AutoLoad;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.WeakHashMap;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import org.bukkit.GameMode;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.plugin.java.JavaPlugin;
+import vg.civcraft.mc.civmodcore.inventory.items.MaterialUtils;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSetting;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSettingAPI;
+import vg.civcraft.mc.civmodcore.players.settings.gui.MenuSection;
+
+public final class SafeOreBreak extends BasicHack {
+
+	@AutoLoad
```

Comment:
You can use `@AutoLoad(processor = DataParser.MATERIAL)` to instead load a `List<Material>` which you can then filter.

Reply:
It's not a list of materials, it's a list of lists of materials

Reply:
Ohhhh, I see, to contain the deepslate variant. Hmmm... is there not a paper API for that? If not there really should be.

Reply:
The advantage is that it's easier for the user, so you need on setting for both deepslate and regular ore

Reply:
Ah right wanting to group multiple blocks together makes sense. I'm not a huge fan of separating multiple entries on one line though, what are your thoughts on formatting it akin too this?
````yaml
ores:
  diamond:
    - DIAMOND_ORE
    - DEEPSLATE_DIAMOND_ORE
  coal:
    - COAL_ORE
    - DEEPSLATE_COAL_ORE
  glass:
    - GLASS
    - GLASS_PANE
```

to create config entries for each (using the first entry as the icon), and expanding the scope of this a bit to make it bit more generic and not just for ores, since there's no harm in providing the option for glass and such. While we're at it, we might as well target everything that you'd want to pick up with silk.

Reply:
> please no scope creep

I agree. Unless those names are going to be used somehow, adding unnecessary structure to the data just adds complexity. It would allow you to use `@AutoLoad(processor = DataParser.MATERIAL)` though, so there's that.